### PR TITLE
f-strings to replace str.format()

### PR DESCRIPTION
**Pull request by cclauss**
Would it be acceptable to move to Python 3.6 and better to get f-string support?

They speed up string concatenation via pre-compilation and a new special bytecode.  Applying f-strings to code generation should speed up that process as well.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -7,12 +7,10 @@
...
 
-VERSION = str(shivyc.__version__)
-DOWNLOAD_URL = ("https://github.com/ShivamSarodia/ShivyC/archive/{}.tar.gz"
-                .format(VERSION))
+f"ShivyC only supports Python 3.6 or later"
 
-if sys.version_info[0] < 3:
-    sys.exit("ShivyC only supports Python 3.")
+VERSION = str(shivyc.__version__)
+DOWNLOAD_URL = f"https://github.com/ShivamSarodia/ShivyC/archive/{VERSION}.tar.gz"
```
Comment (ShivamSarodia) [inline setup.py]: Curious - why remove the Python version check here?

**Comment by ShivamSarodia** [comment]
Thanks for this PR - it really does clean up the code :) See my comment above, and also, could you modify the first paragraph in the Quickstart section of the README from "ShivyC requires only Python 3" to "ShivyC requires only Python **3.6**"?

By the way, I've added some current TODOs under the Issues tab (labeled as Features). If you'd like to continue your contributions to ShivyC, feel free to tackle any that look interesting. I'm happy to help out if you have questions about any parts of the codebase.

File: setup.py (hunk 1 above)
Comment (cclauss) [reply setup.py]: On any Python version that does not support f-strings the result of the new line above will be:
```
>>> +f"ShivyC only supports Python 3.6 or later"
  File "<string>", line 1
    +f"ShivyC only supports Python 3.6 or later"
                                               ^
SyntaxError: invalid syntax
```

Comment (ShivamSarodia) [reply setup.py]: Ah, I see! Thanks.

**Pull request merged**